From 9cf946fb274be730185e83cf94b050cfe839358f Mon Sep 17 00:00:00 2001
From: Max <max.maisel@posteo.de>
Date: Thu, 1 Mar 2018 17:27:14 +0100
Subject: [PATCH 4/6] ext4_blockdev: Added user data pointer

---
 include/ext4_blockdev.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/ext4_blockdev.h b/include/ext4_blockdev.h
index 41223b6..4efbb43 100644
--- a/include/ext4_blockdev.h
+++ b/include/ext4_blockdev.h
@@ -97,6 +97,9 @@ struct ext4_blockdev_iface {
 
 	/**@brief   Physical write counter*/
 	uint32_t bwrite_ctr;
+
+	/**@brief   User data pointer*/
+	void* p_user;
 };
 
 /**@brief   Definition of the simple block device.*/
-- 
2.7.4

